apps/sms/js/drafts.js

```diff
@@ -32,183 +26,107 @@
     get size() {
       return draftIndex.size;
     },
+
     /**
-     * add
-     *
-     * Push a Draft object or an object that has
-     * all the properties of a Draft instance
-     * to the Drafts collection. If the object
-     * isn't an instance of Draft, initialize
-     * a new Draft.
-     *
-     * @param  {Object}  draft  Draft-like object.
-     *
-     * @return {Drafts} return the Drafts object.
+     * Pushes a Draft instance to the Drafts in-memory collection and commits
+     * it to the persistent storage.
+     * @param  {Draft} draft Draft instance.
+     * @returns {Drafts} Returns the Drafts object.
      */
     add: function(draft) {
-      var threadId;
-      var thread;
-      var stored;
-
-      if (draft) {
-        if (!(draft instanceof Draft)) {
-          draft = new Draft(draft);
-        }
+      if (!draft || !(draft instanceof Draft)) {
```

Comment:
While trying to understand Drafts code I found it difficult to read/follow/understand when ```add/delete``` accepts different types of objects, so I'd love to be consistent here and accept only one type - ```Draft```, but what do you think?